nntrainer/tensor/tensor.cpp

```diff
@@ -1874,14 +1874,22 @@ Tensor &Tensor::sum(unsigned int axis, Tensor &ret, float alpha,
         unsigned int t_axis = dim[2];
         Tensor ones(1, 1, 1, t_axis, this->getTensorType());
         ones.setValue(alpha);
-        float *rdata = ret.getData<float>();
-        for (unsigned int k = 0; k < dim[0]; ++k) {
-          for (unsigned int c = 0; c < dim[1]; ++c) {
-            unsigned int idx = k * dim.getFeatureLen() + c * dim[3] * dim[2];
-            unsigned int ridx = k * ret.dim.getFeatureLen() + c * dim[3];
-            sgemv(CblasRowMajor, CblasTrans, t_axis, t_3, 1, &data[idx], t_3,
-                  ones.getData<float>(), 1, beta, &rdata[ridx], 1);
+
+        if (dim.getStorageOrder() == TStorageOrder::ROW_MAJOR) {
+          float *rdata = ret.getData<float>();
```

Comment:
It is very similar to the code in line 1923.
How about creating a separate function?

Reply:
I see there are a total of 18 cases in this function (250 lines of code!). I'll figure out ways to minimize this in TensorV2 whether is utilizing a function or macros.